### PR TITLE
Compilers rfc

### PR DESCRIPTION
**Pull request by VoR0220**
This contains the compilers RFC which dockerizes the ethereum solc compiler and utilizes it as a service and operates it mostly from the CLI.

### CODE REVIEW

File: initialize/default_files.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ const tomlHeader = `# This is a TOML config file.
 // ------------------ services ------------------
 
 var ServiceDefinitions = []string{
+	"solc"
```
Comment (zramsay) [inline initialize/default_files.go]: the missing `,` prevents the code as is from compiling.

File: initialize/initialize.go [hunk 2]
```diff
@@ -140,13 +143,15 @@ func pullDefaultImages(images []string) error {
...
+		"solc":      version.ImageSolc,
 	}
 
 	for i, image := range images {
 		images[i] = versionedImageNames[image]
 
 		// Attach default registry prefix.
-		if !strings.HasPrefix(images[i], version.DefaultRegistry) {
+		// [rj] - this is very hacky, need to fix this either in images.go via a map, struct or a function
+		if !strings.HasPrefix(images[i], version.DefaultRegistry) && image != "solc" {
```
Comment (zramsay) [inline initialize/initialize.go]: it works though. `monax init` did exactly what it should've

File: version/images.go [hunk 3]
```diff
@@ -20,4 +21,5 @@ var (
 	ImageKeys      = fmt.Sprintf("monax/keys:%s", KEYS_VERSION)
 	ImageDB        = fmt.Sprintf("monax/db:%s", DB_VERSION)
 	ImageCompilers = fmt.Sprintf("monax/compilers:%s", COMPILERS_VERSION)
```
Comment (zramsay) [inline version/images.go]: ummm, this needs to be deprecated.

**Comment by zramsay** [comment]
the compilers tests also leave my git tree dirty

File: initialize/default_files.go (hunk 1 above)
Comment (VoR0220) [inline initialize/default_files.go]: Shit. Sorry bout that.

File: initialize/initialize.go (hunk 2 above)
Comment (VoR0220) [inline initialize/initialize.go]: It does. But something to clean in the future.

File: version/images.go (hunk 3 above)
Comment (VoR0220) [inline version/images.go]: ++

**Comment by VoR0220** [comment]
Ah shit...yea i will fix this.

**Comment by VoR0220** [comment]
Monax/cli/compilers is a new package to maintain compilers. The monax/compilers needs to be rid from the vendor however. Good catch.

File: version/images.go (hunk 3 above)
Comment (VoR0220) [inline version/images.go]: Well it will be deprecated. Thats for the second PR.

Comment (zramsay) [inline version/images.go]: what, why? `ethereum/solc` is *replacing* `quay.io/monax/compilers` ... that's definitely for one PR

Comment (VoR0220) [inline version/images.go]: There's a way ive created the compiler interaction with the deploy job that is for the second portion.

**Pull request closed without merging**